### PR TITLE
Implement expiration mode on `Config`

### PR DESCRIPTION
**Pull request by zenangst**
Implements a new configuration option for opting out from using the internal expiration cleanup mechanism in `Cache`.

Expiration mode lives on `Config` and is used by `CacheManager`. When the expiration mode is set to `.auto`, `Cache` will handle the cleanup of expired objects. When set to `.manual`, it means that you will manually handle expired objects.

This should be a non-breaking change as `.auto` is the default.

Attempting to solve: https://github.com/hyperoslo/Cache/issues/116

### CODE REVIEW

File: Tests/iOS/Tests/Cache/SpecializedCacheTests.swift [hunk 1]
```diff
@@ -33,6 +36,7 @@ final class SpecializedCacheTests: XCTestCase {
   // MARK: - Async caching
 
   func testAsyncAddObject() throws {
+    cache = SpecializedCache<User>(name: cacheName)
```
Comment (vadymmarkov) [inline Tests/iOS/Tests/Cache/SpecializedCacheTests.swift]: Why did you move it here instead of `setup`?

Comment (zenangst) [reply Tests/iOS/Tests/Cache/SpecializedCacheTests.swift]: Had some issues with async calls from the `setup` method clearing the cache even if the test shouldn't do that.

Comment (zenangst) [reply Tests/iOS/Tests/Cache/SpecializedCacheTests.swift]: But I kinda don't want to leave it like this, the tests feel a bit icky now :/

**Comment by onmyway133** [comment]
@zenangst I just think that to manually handle cache, just set an expiry date to 100 years from now 🤔

**Comment by zenangst** [comment]
@vadymmarkov when `Cache` clears a cache, will it remove the entire directory or only the cached objects assigned to that specific cache container?

**Comment by zenangst** [comment]
@onmyway133 but then you don't know if they are expired or not. What we are trying to achieve here is to decouple the expiration date and the automatic cleanup. Giving the developer more control, maybe you want it to clear when it is expired but not in the scenario when the phone is in offline.

**Comment by vadymmarkov** [comment]
@zenangst `clear` method can do both, there is a parameter for that.

**Comment by zenangst** [comment]
I think this implementation can be improved!
